src/evm.rs

```diff
@@ -744,6 +770,67 @@ impl<'a, Ext, Db: Database + DatabaseCommit, TrevmState: NeedsBlock>
         }
     }
 
+    /// Open a block, apply some logic, and return the EVM ready for the next
+    /// block.
+    pub fn drive_block<'b, B, C>(self, driver: &'b B) -> DriveBlockResult<'a, 'b, Ext, Db, C, B>
+    where
+        C: BlockContext,
+        B: BlockDriver<'b, C>,
+    {
+        let trevm = self
+            .open_block(driver.block(), driver.context())
+            .map_err(EvmErrored::err_into::<<B as BlockDriver<'b, C>>::Error<Db>>)?;
+        let trevm = driver.run_txns(trevm)?;
+
+        let trevm = trevm.close_block().map_err(EvmErrored::err_into)?;
+
+        match driver.post_block_checks(&trevm) {
```

Comment:
I think we should mark this as a TODO too indicating it's a noop atm.

Reply:
#30